gaiku-common/src/data/chunk.rs

```diff
@@ -1,97 +1,125 @@
 use mint::{Vector3, Vector4};
 
-// TODO: Get inspiration on multiarray crate (https://github.com/sellibitze/multiarray) to make chunk 2d and 3d friendly
+pub trait Chunkify {
+  fn new(position: [f32; 3], width: usize, height: usize, depth: usize) -> Self;
+  fn depth(&self) -> usize;
+  fn is_air(&self, x: usize, y: usize, z: usize) -> bool;
+  fn get(&self, x: usize, y: usize, z: usize) -> u8;
+  fn height(&self) -> usize;
+  fn position(&self) -> Vector3<f32>;
+  fn width(&self) -> usize;
+}
+
+pub trait ChunkifyMut {
+  fn set(&mut self, x: usize, y: usize, z: usize, value: u8);
+}
```

Comment:
@QuantumEntangledAndy something like this is what are you looking for?

Reply:
That's good it will make it clear at compile time if a method needs a chunk that has to be set